src/App.tsx

```diff
@@ -1,39 +1,68 @@
-import React from 'react';
+import { Component } from 'react';
+import { Clock } from './Clock';
 import './App.scss';
 
+type State = {
+  hasClock: boolean,
+  clockName: string,
+  previousClockName: string,
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+    previousClockName: '',
+  };
+
+  timerForClock = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleContextMenu);
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.timerForClock = window.setInterval(() => {
+      this.setState(prevState => ({
+        clockName: getRandomName(),
+        previousClockName: prevState.clockName,
+      }));
+    }, 3300);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    clearInterval(this.timerForClock);
+  }
```

Comment:
Remove event listeners for 'contextmenu' and 'click' events on componentWillUnmount.